bagelbids/cli.py

```diff
@@ -193,32 +221,38 @@ def bids(
     for bids_sub_id in layout.get_subjects():
         pheno_subject = pheno_subject_dict.get(f"sub-{bids_sub_id}")
         session_list = []
+
+        bids_sessions = layout.get_sessions(subject=bids_sub_id)
+        if not bids_sessions and layout.get_datatypes(subject=bids_sub_id):
```

Comment:
:cherries:: I think `layout.get_datatypes(subject=bids_sub_id)` is not needed here, right? Because if there aren't any datatypes, then the return `image_list` will just be empty anyway.

Reply:
That's true, but the inclusion of this condition means that when there are no datatypes, the execution of the below for loop + `create_acquisitions()` are skipped altogether. That's ideal right?

Reply:
Oh, I see, you want to just skip all that. I had missed that on the first read. I think I would do this more explicitly then, like so:

```python
if not bids_sessions:
  if not layout.get_datatypes(subject=bids_sub_id):
    continue
  else:
    bids_sessions = [None]
```
That said: I'm not convinced that we need to optimize for performance here, or that what you're proposing would do that. I assume that the slow part of all this code is calling the `layout` methods (like `layout.get` or probably also `layout.get_datatypes`). Your version occasionally skips one `layout.get` call, but always adds a new `layout.get_datatypes` call, so the net benefit might very well be negative. 

So my personal vote would be on just removing it and using the check for empty `image_list` below to handle this case.